db/migrate/20170909151007_add_title_to_versions.rb

```diff
@@ -0,0 +1,8 @@
+class AddTitleToVersions < ActiveRecord::Migration[5.1]
+  def change
+    add_column :versions, :title, :string
+    Version.includes(:page).find_each do |v|
+      puts "Updated Version record with UUID #{v.uuid} to have title '#{v.title}'" if v.update(title: v.page.title)
```

Comment:
I think this is going to be verbose enough to be a problem (we have ~335,000 versions in production right now). This much logging will actually slow the migration way down and probably not be helpful. Maybe better to count the number of changes and just output a summary, e.g. “Updated title on X version records for Y pages.”

Also, this should probably only run when migrating up, not down, so maybe move this part into a `reversible` block with only an `up` section?

Finally, Rails has some nice logging helpers for migrations. You could do something like:

```rb
say_with_time("Copying titles from pages to versions") do
  Version.includes(:page).find_each do |v|
    # Update version and count it
  end
  say("X versions", :subitem)
  say("Y pages", :subitem)
end
```

…and you get nice output like:

```
== 20170909151007 AddTitleToVersions: migrating ===============================
-- add_column(:versions, :title, :string)
   -> 0.0013s
-- Copying titles from pages to versions
   -> 1019 versions
   -> 146 pages
   -> 2.3985s
== 20170909151007 AddTitleToVersions: migrated (2.3999s) ======================
```

Reply:
Oh, also: this might be faster if you roll through the list of pages and call `update_all` on their versions; that way there’s only one SQL query per page, rather than one for every version, e.g:

```rb
Page.each do |page|
  page.versions.update_all(title: page.title)
end
```

`update_all` also returns the number of records that were affected, which is helpful in doing the summary counting I mentioned above.

Reply:
Very cool! I've never made use of these Rails helpers before, so TIL. I'll change that up.